druid-shell/src/platform/mac/window.rs

```diff
@@ -168,11 +168,16 @@ impl WindowBuilder {
 
             if let Some(min_size) = self.min_size {
                 let frame = NSWindow::frame(window);
-                let content_height = window.contentRectForFrameRect_(frame).size.height;
-                let window_height = frame.size.height;
-                let title_bar_height = window_height - content_height;
-                // setMinSize_ does not take into account the height of the title bar.
-                let size = NSSize::new(min_size.width, min_size.height + title_bar_height);
+                let content_dimensions = window.contentRectForFrameRect_(frame).size;
+                let excess_height = frame.size.height - content_dimensions.height;
+                let excess_width = frame.size.width - content_dimensions.width;
+
+                // setMinSize_ restricts the size of the window including padding.
+                let size = NSSize::new(
+                    min_size.width + excess_width,
+                    min_size.height + excess_height,
+                );
+
                 window.setMinSize_(size);
```

Comment:
Looking at the NSWindow docs there is actually a method for what I am trying to achieve here. SetContentMinSize_.